Mathlib/Order/Monotone/Basic.lean

```diff
@@ -1104,3 +1105,26 @@ alias ⟨Monotone.apply₂, Monotone.of_apply₂⟩ := monotone_iff_apply₂
 alias ⟨Antitone.apply₂, Antitone.of_apply₂⟩ := antitone_iff_apply₂
 
 end apply
+
+/-- A monotone function `f : ℕ → ℕ` bounded by `B`, which is constant after stabilising for the
+first time, stabilises in at most `B` steps. -/
+lemma Nat.stabilises_of_monotone {f : ℕ → ℕ} {b n : ℕ} (hfmono : Monotone f) (hfb : ∀ m, f m ≤ b)
+    (hfstab : ∀ m, f m = f (m + 1) → f (m + 1) = f (m + 2)) (hbn : b ≤ n) : f n = f b := by
+  obtain ⟨m, hmb, hm⟩ : ∃ m ≤ b, f m = f (m + 1) := by
+    contrapose! hfb
+    let rec strictMono : ∀ m ≤ b + 1, m ≤ f m
+    | 0, _ => Nat.zero_le _
+    | m + 1, hmb => (strictMono _ <| m.le_succ.trans hmb).trans_lt <| (hfmono m.le_succ).lt_of_ne <|
+        hfb _ <| Nat.le_of_succ_le_succ hmb
+    exact ⟨b + 1, strictMono _ le_rfl⟩
+  replace key : ∀ k : ℕ, f (m + k) = f (m + k + 1) ∧ f (m + k) = f m := fun k =>
+    Nat.rec ⟨hm, rfl⟩ (fun k ih => ⟨hfstab _ ih.1, ih.1.symm.trans ih.2⟩) k
+  replace key : ∀ k ≥ m, f k = f m := fun k hk =>
+    (congr_arg f (Nat.add_sub_of_le hk)).symm.trans (key (k - m)).2
+  exact (key n (hmb.trans hbn)).trans (key b hmb).symm
+
+@[deprecated (since := "2024-11-27")]
+alias Group.card_pow_eq_card_pow_card_univ_aux := Nat.stabilises_of_monotone
+
+@[deprecated (since := "2024-11-27")]
+alias Group.card_nsmul_eq_card_nsmulpow_card_univ_aux := Nat.stabilises_of_monotone
```

Comment:
Seems like we should try to automatically detect if to_additive didn't modify the statement

Reply:
I am not convinced yet that there isn't a legitimate use for it, but yes good idea